### ISSUE TITLE
 undefined method `s' for DockerCookbook::DockerNetwork

### ISSUE THREAD

**Issue by rmoriz** [chef-cookbooks/docker#890]
There is a Bug in DockerCookbook::DockerNetwork which seems to occur if a network will be deleted/recreated and no explicit gateway was specified.

The network was created with chef and docker_network but attributes were changed. On the very next run chef-client fails:

Error:

 ```
 NoMethodError
 -------------
 undefined method `s' for DockerCookbook::DockerNetwork
 
 Cookbook Trace:
 ---------------
 /var/chef/cache/cookbooks/docker/libraries/helpers_network.rb:95:in `block in consolidate_ipam'
 /var/chef/cache/cookbooks/docker/libraries/helpers_network.rb:84:in `each'
 /var/chef/cache/cookbooks/docker/libraries/helpers_network.rb:84:in `consolidate_ipam'
 /var/chef/cache/cookbooks/docker/libraries/docker_network.rb:69:in `block (3 levels) in <class:DockerNetwork>'
 /var/chef/cache/cookbooks/docker/libraries/helpers_base.rb:66:in `with_retries'
 /var/chef/cache/cookbooks/docker/libraries/docker_network.rb:65:in `block (2 levels) in <class:DockerNetwork>'
 /var/chef/cache/cookbooks/docker/libraries/docker_network.rb:62:in `block in <class:DockerNetwork>'
 
 Resource Declaration:
 ---------------------
  47: docker_network network_name do
  48:   enable_ipv6 true
  49:   subnet 'fd00:dead:beef::/48'
  50:   action :create
  51: end
  52: 
```

Probably caused by:

https://github.com/chef-cookbooks/docker/blob/bcfaa3c3e52514ceddd0128d07fdaee23bf1b63b/libraries/helpers_network.rb#L86-L95

`s` is not available in the raised exception message.

One can also argue about the need to raise an exception here, too: Internal networks usually don't have a gateway.

**Comment by chasebolt** [comment]
the `gateways.each` block shouldnt be ran if gateways is an empty array,

**Comment by chasebolt** [comment]
resolved via #889
